.github/workflows/plantuml.yml

```diff
@@ -0,0 +1,39 @@
+name: Generate PlantUML Diagrams
+
+on:
+  pull_request:
+    paths:
+      - "**.puml"
+  push:
+    paths:
+      - "**.puml"
+    branches:
+      - master
```

Comment:
Do we really want to trigger on master push. diagram should anyway be updated in the PR before being merged and we should not (auto-commit) to master directly?

Reply:
for the same reason you generally want to run tests again on the default branch after they already run on a PR - if multiple PRs get worked in parallel and get merged as they are ready it's always best to also run in master to avoid inconsistencies after merging the N+1 PR

Reply:
and of course this workflow would just run when a commit contains puml files changes and it's pretty fast to run